File: apps/party-time-backend/src/main/java/com/partytime/jpa/DatabaseConstants.java
package com.partytime.jpa;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class DatabaseConstants {

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static final class Account {
        public static final String TABLE_NAME = "ACCOUNT";
        public static final String COLUMN_EMAIL = "EMAIL";
        public static final String COLUMN_EMAIL_VERIFIED = "EMAIL_VERIFIED";
        public static final String COLUMN_NAME = "NAME";
        public static final String COLUMN_PW_HASH = "PASSWORD_HASH";
        public static final String COLUMN_EMAIL_VERIFICATION_CODE = "EMAIL_VERIFICATION_CODE";
        public static final String COLUMN_PASSWORD_VERIFICATION_CODE = "PASSWORD_VERIFICATION_CODE";
    }

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static final class Event {
        public static final String TABLE_NAME = "EVENT";
        public static final String COLUMN_ORGANIZER_ID = "ORGANIZER_ID";
        public static final String COLUMN_NAME = "NAME";
        public static final String COLUMN_DATE_TIME = "DATE_TIME";
        public static final String COLUMN_ADDRESS = "ADDRESS_ID";
    }

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static final class EventParticipant {
        public static final String TABLE_NAME = "EVENT_PARTICIPANT";
        public static final String COLUMN_ACCOUNT_ID = "ACCOUNT_ID";
        public static final String COLUMN_EVENT_ID = "EVENT_ID";
        public static final String COLUMN_STATUS = "STATUS";
    }

    @NoArgsConstructor(access = AccessLevel.PRIVATE)
    public static final class Address {
        public static final String TABLE_NAME = "ADDRESS";
        public static final String COLUMN_ADDRESS_LINE = "ADDRESS_LINE";
        public static final String COLUMN_ZIP = "ZIP";
        public static final String COLUMN_CITY = "CITY";
        public static final String COLUMN_COUNTRY = "COUNTRY";
    }

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/jpa/DatabaseConstants.kt
package com.partytime.jpa

object DatabaseConstants {
    object Account {
        const val TABLE_NAME: String = "ACCOUNT"
        const val COLUMN_EMAIL: String = "EMAIL"
        const val COLUMN_EMAIL_VERIFIED: String = "EMAIL_VERIFIED"
        const val COLUMN_NAME: String = "NAME"
        const val COLUMN_PW_HASH: String = "PASSWORD_HASH"
        const val COLUMN_EMAIL_VERIFICATION_CODE: String = "EMAIL_VERIFICATION_CODE"
        const val COLUMN_PASSWORD_VERIFICATION_CODE: String = "PASSWORD_VERIFICATION_CODE"
    }

    object Event {
        const val TABLE_NAME: String = "EVENT"
        const val COLUMN_ORGANIZER_ID: String = "ORGANIZER_ID"
        const val COLUMN_NAME: String = "NAME"
        const val COLUMN_DATE_TIME: String = "DATE_TIME"
        const val COLUMN_ADDRESS: String = "ADDRESS_ID"
    }

    object EventParticipant {
        const val TABLE_NAME: String = "EVENT_PARTICIPANT"
        const val COLUMN_ACCOUNT_ID: String = "ACCOUNT_ID"
        const val COLUMN_EVENT_ID: String = "EVENT_ID"
        const val COLUMN_STATUS: String = "STATUS"
    }

    object Address {
        const val TABLE_NAME: String = "ADDRESS"
        const val COLUMN_ADDRESS_LINE: String = "ADDRESS_LINE"
        const val COLUMN_ADDRESS_LINE_ADDITION: String = "ADDRESS_LINE_ADDITION"
        const val COLUMN_ZIP: String = "ZIP"
        const val COLUMN_CITY: String = "CITY"
        const val COLUMN_COUNTRY: String = "COUNTRY"
    }
}
